src/coreclr/jit/block.h

```diff
@@ -551,6 +551,8 @@ enum BasicBlockFlags : unsigned __int64
     BBF_HAS_ALIGN            = MAKE_BBFLAG(39), // BB ends with 'align' instruction
     BBF_TAILCALL_SUCCESSOR   = MAKE_BBFLAG(40), // BB has pred that has potential tail call
 
+    BBF_BACKWARD_JUMP_SOURCE = MAKE_BBFLAG(41), // Block is a source of a backward jump
```

Comment:
It's unfortunate that we have/need bits for this (and BBF_BACKWARD_JUMP_TARGET). Lots of code just compares bbNum, but that depends on bbNum being up-to-date; presumably you can't depend on that? Oh, I guess in addition, you don't want to depend on the preds list being up-to-date and walking the preds list? These bits can obviously themselves get out of date if the flow graph is reordered. Also, do we properly update these bits when, say, we merge blocks and delete the one with the bit?

Reply:
The def/use happens very early, in the flow graph build and importer, before anything else can mess with the flow graph. I could look at source/target nums instead, but that would require enumeration of succs (for sources) and preds (for targets) for every block which would be more costly, and I'd have to build cheap preds.

Nothing later depends on these flag bits. I could scrub them out (at least in debug) if you think leaving them around and potentially gone stale is an attractive nuisance.

Or we could figure out how to have limited-lifetime flags that automatically expire at some point.

Reply:
It would be useful if we knew when any of the block bits were expected to be valid, either as documentation here, or better, with some kind of checking. Scrubbing them might not be possible (they're only 0/1 already, so scrubbing to 0 doesn't help), unless we add a "valid/invalid" bit for each one, with a debug accessor/checker. Certainly not required with this change (unless you wanted to add comments describing their lifetime).